frontend/public/components/modals/cluster-update-modal.tsx

```diff
@@ -126,41 +138,87 @@ const ClusterUpdateModal = withHandlePromise((props: ClusterUpdateModalProps) =>
+        </>
+      ) : (
+        version
+      ),
+    };
+  };
+  const recommendedOptions = availableSortedUpdates.map(({ version }) => {
+    return dropdownItem(version);
+  });
+  const notRecommendedOptions = notRecommendedSortedUpdates.map(({ release: { version } }) => {
+    return dropdownItem(version);
   });
+  const options = [
+    {
+      label: t('public~Recommended'),
+      items: recommendedOptions,
+    },
+  ];
+  if (includeNotRecommended) {
+    options.unshift({
+      label: t('public~Supported but not recommended'),
+      items: notRecommendedOptions,
+    });
+  }
   const helpLink = isUpstream()
     ? `${openshiftHelpBase}updating_clusters/update-using-custom-machine-config-pools.html`
     : `${openshiftHelpBase}html/updating_clusters/update-using-custom-machine-config-pools.html`;
-  const { t } = useTranslation();
 
   return (
     <form onSubmit={submit} name="form" className="modal-content modal-content--no-inner-scroll">
       <ModalTitle>{t('public~Update cluster')}</ModalTitle>
       <ModalBody>
-        {clusterUpgradeableFalse && <ClusterNotUpgradeableAlert cv={cv} />}
+        {clusterUpgradeableFalse && <ClusterNotUpgradeableAlert cancel={cancel} cv={cv} />}
         <div className="form-group">
           <label>{t('public~Current version')}</label>
           <p>{currentVersion}</p>
         </div>
         <div className="form-group">
           <label id="version-label">{t('public~Select new version')}</label>
-          <Select
-            aria-labelledby="version-label"
-            onToggle={onToggleVersion}
+          <DropdownWithSwitch
+            isFullWidth
             onSelect={onSelectVersion}
-            selections={desiredVersion}
-            isOpen={isOpen}
-            isDisabled={clusterUpgradeableFalse && !currentMinorVersionPatchUpdate}
-          >
-            {options}
-          </Select>
+            options={options}
+            selected={desiredVersion}
+            switchIsChecked={includeNotRecommended}
+            switchIsDisabled={notRecommendedOptions.length === 0}
+            switchLabel={t('public~Include supported but not recommended versions')}
+            switchLabelClassName="co-switch-label"
+            switchLabelIsReversed
+            switchOnChange={(val) => setIncludeNotRecommended(val)}
+            toggleLabel={desiredVersion}
+          />
+          {desiredNotRecommended && (
+            <Alert
+              className="pf-u-mt-sm"
+              isExpandable
+              isInline
+              title={t(
+                'public~Updating this cluster to {{desiredVersion}} is supported, but not recommended as it might not be optimized for some components in this cluster.',
+                { desiredVersion: desiredNotRecommended.release.version },
+              )}
+              variant="info"
+            >
+              <TextContent>
+                {desiredNotRecommended.risks.map((risk) => (
```

Comment:
This should show condition `reason` and `message` instead of looping the risks?  @wking

Reply:
https://github.com/openshift/console/pull/11424/commits/a698ca92566a8f8a382e75a3c818f444c91df56d changes to use the condition `message` which is what I think we want, but will let Trevor confirm.

Reply:
Yeah, the cluster-version operator is already iterating over risks and generating the consolidated recommended condition to summarize all risks it is concerned about.